src/components/tasks/card/index.tsx

```diff
@@ -354,6 +425,88 @@ const Card: FC<CardProps> = ({
         };
     };
 
+    const onCancelEditRoute = () => {
+        setIsEditMode(false);
+    };
+
+    const CancelEditButton = () => (
+        <div className={classNames.cancelEditButton}>
+            <Image
+                src="/cancel.png"
+                alt="cancel icon to close edit mode"
+                width={20}
+                height={20}
+                onClick={onCancelEditRoute}
+                tabIndex={0}
+                data-testid="cancel-edit-button"
+            />
+        </div>
+    );
+
+    const handleTitleChange = (value: string) => {
+        setEditedTaskDetails((prev) => ({
+            ...prev,
+            savingText: PENDING,
+        }));
+
+        const response = updateTask({
+            id: cardDetails.id,
+            task: {
+                title: value,
+            },
+        });
+
+        response
+            .unwrap()
+            .then((result) => {
+                setEditedTaskDetails((prev) => ({
+                    ...prev,
+                    savingText: SAVED,
+                }));
+            })
+            .catch((err) => {
+                toast(ERROR, err.data.message);
+            })
+            .finally(() => {
+                setTimeout(() => {
+                    setEditedTaskDetails((prev) => ({
+                        ...prev,
+                        savingText: '',
+                    }));
+                }, 3000);
+            });
+    };
+
+    const debounceTitle = (func: (value: string) => void, delay: number) => {
+        const timerIdRef = useRef<NodeJS.Timeout | undefined>();
+        return (value: string) => {
+            if (timerIdRef.current) {
+                clearTimeout(timerIdRef.current);
+                setEditedTaskDetails((prev) => ({
+                    ...prev,
+                    savingText: '',
+                }));
+            }
+            timerIdRef.current = setTimeout(() => {
+                func(value);
+            }, delay);
+        };
+    };
+
+    const debouncedHandleTitleChange = debounceTitle(handleTitleChange, 3000);
```

Comment:
inputHandler function is supposed to update the task title and call the debouncedHandleTitleChange function. However, the inputHandler function is not checking to make sure that the value of the input element is not empty. This means that if the user enters an empty value into the input element, the debouncedHandleTitleChange function will not be called and the task title will not be updated.

Reply:
I have fixed it now, if user tries to enter the empty value then it will notify the user showing toast that “Title is not allowed to be empty”.